src/sage/rings/semirings/tropical_variety.py

```diff
@@ -0,0 +1,1124 @@
+                    # calculate order
+                    index_diff = []
+                    for i in range(len(keys[0])):
+                        index_diff.append(abs(keys[0][i]-keys[1][i]))
+                    order = gcd(index_diff)
+                    temp_order.append(order)
+                    temp_keys.append(keys)
+
+        # changing all the operator symbol to be <= or >=
+        self._keys = []
+        components = []
+        dim_param = len(tropical_roots[0][0]) - 1
+        vars = [SR.var('t{}'.format(i)) for i in range(1, dim_param+1)]
+        for arg in tropical_roots:
+            subs_dict = {}
+            index_vars = 0
+            new_eq = []
+            for eq in arg[0]:
+                var_eq = eq.variables()
+                for var in var_eq:
+                    if var not in subs_dict:
+                        subs_dict[var] = vars[index_vars]
+                        index_vars += 1
+                new_eq.append(eq.subs(subs_dict))
+            new_eq = tuple(new_eq)
+            arg.remove(arg[0])
+            arg.insert(0, new_eq)
+            if not arg[1]:
+                for var in vars:
+                    expr1 = -infinity < var
+                    expr2 = var < infinity
+                    arg[1].append(expr1)
+                    arg[1].append(expr2)
+            else:
+                params = arg[1]
+                arg.remove(params)
+                new_param = []
+                for param in params:
+                    lhs = param.lhs().subs(subs_dict)
+                    rhs = param.rhs().subs(subs_dict)
+                    if param.operator() == operator.gt:
+                        expr = lhs >= rhs
+                    else:
+                        expr = lhs <= rhs
+                    new_param.append(expr)
+                arg.insert(1, new_param)
+            components.append(arg)
+
+        # determine the order of each component
+        self._vars = vars
+        final_order = []
+        for i, component in enumerate(components):
+            if component not in self._hypersurface:
+                self._hypersurface.append(component)
+                final_order.append(temp_order[i])
+                self._keys.append(temp_keys[i])
+            else:
+                index = self._hypersurface.index(component)
+                if temp_order[i] > final_order[index]:
+                    final_order[index] = temp_order[i]
+                    self._keys[index] = temp_keys[i]
+        for i in range(len(self._hypersurface)):
+            self._hypersurface[i].append(final_order[i])
+
+    def dimension(self):
+        """
+        Return the dimension of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<a,x,y,z> = PolynomialRing(T)
+            sage: p1 = x*y + R(-1)*x*z
+            sage: p1.tropical_variety().dimension()
+            4
+        """
+        return len(self._hypersurface[0][0])
```

Comment:
```suggestion
        return ZZ(len(self._hypersurface[0][0]))
```
Avoid return Python `int`s.